dds/DCPS/Serializer.inl

```diff
@@ -783,7 +783,7 @@ bool Serializer::align_r(size_t al)
   }
   al = (std::min)(al, encoding().max_align());
   const size_t len =
-    (al - ptrdiff_t(current_->rd_ptr()) + align_rshift_) % al;
+    (al - reinterpret_cast<size_t>(current_->rd_ptr()) + align_rshift_) % al;
```

Comment:
This previously used `ptrdiff_t` since that's an integer type guaranteed to be as wide as the pointer type.  For portability / future compatibly I'd like to maintain that property.  Is there a way to use `ptrdiff_t` and avoid warnings?

Reply:
The problem is that `al` is unsigned so whatever `current->rd_ptr()` is cast to needs to be unsigned to prevent the warning.

Reply:
Maybe the whole expression needs a re-write?  It really only matters what the low few bytes of the rd_ptr are (which is `rd_ptr % al` with `al` being a power of two.